### ISSUE TITLE
Referrer override doesn't work

### ISSUE THREAD

**Issue by gyril** [segment-integrations/analytics.js-integration-google-analytics#22]
As per the GA doc, to override the referrer of a page (which defaults to `document.referrer`), one must use `ga('set', 'referrer', new_referrer)`.

As per the Segment doc, this result can be obtained by passing properties to the `page()` call, as such: `analytics.page({referrer: new_referrer})`

However, this has no effect. The bug I believe comes from the fact that there is no `.option('referrer', document.referrer)` declared in the GA lib (https://github.com/segment-integrations/analytics.js-integration-google-analytics/blob/master/lib/index.js#L51), which causes the `metrics()` call to return nothing and doesn't issue a `ga('set')` call.

**Comment by gyril** [comment]
It actually shouldn't have anything to do with `metrics()` which is just for custom metrics & dimensions. It should be set along with the `pagePath` and `pageTitle`

**Comment by hankim813** [comment]
@gyril do you have links to ga docs where it speaks about this? I think the option to override `referrer` was left out for a reason but def worth revisiting since if ga makes it clear that this is okay to do, then we should map this

**Comment by gyril** [comment]
Hi @hankim813 , as per [1] it is documented. I tested it and it works, you can change the referrer for a GA session, mid session, and the change will be reflected in the real-time report.

[1] https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#referrer

**Comment by hankim813** [comment]
I'm trying to find this in GA's doc but @gyril do you know if we can set all three in one call? 

``` javascript
window.ga('set', { page: pagePath, title: pageTitle, referrer: referrer });
```

**Comment by gyril** [comment]
Yes apparently the `set` call can take an object as parameter [0]

[0] https://developers.google.com/analytics/devguides/collection/analyticsjs/command-queue-reference#set

**Comment by hankim813** [comment]
So my only concern here is that if we implement this, it will always override the referrer. Now because our `.page()` call already collects referrer, there's no way to only set the referrer when someone explicitly passes in a referrer:

```
analytics.page({ referrer: 'xyz.com' });
```

And I'm not 100% certain that GA's own under-the-hood calculation for referrer is the same as our own library's.. 

Would be interested in hearing your thoughts on that specific matter @f2prateek @sperand-io

**Comment by gyril** [comment]
Well @sperand-io arrived to the same conclusion 16 days ago [0]. He wanted to implement it defensively, so as not to override GA's action for every single call. I don't know if he's had any success so far.

[0] https://github.com/segment-integrations/analytics.js-integration-google-analytics/pull/23

**Comment by hankim813** [comment]
Fixed in #25
